cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/ProvisionerUtils.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.detector;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.NewTopic;
+
+import static com.linkedin.kafka.cruisecontrol.KafkaCruiseControlUtils.maybeIncreasePartitionCount;
+
+/**
+ * A util class for provisions.
+ */
+public final class ProvisionerUtils {
+
+  private ProvisionerUtils() {
+  }
+
+  /**
+   * Determine the status of increasing the partition count
+   *
+   * @param adminClient AdminClient to handle partition update requests.
+   * @param provisionerTopic Existing topic to add more partitions if needed
+   * @return The state COMPLETE when true, COMPLETED_WITH_EXCEPTION when false
```

Comment:
Since we are using the `COMPLETED` state to categorize the `NO_ACTION` option, the `topicDescription.partitions().size() >= topicToAddPartitions.numPartitions()` will still return true. 
There are currently a few other methods that rely on the `maybeIncreasePartitionCount` method. To limit the scope of this change, we have the `ProvisionerUtils#partitionIncreaseStatus` method wrapping the results. As a future improvement, we could still modify the `maybeIncreasePartitionCount` method to return an enum of the `ProvisionerState.State` or a `ProvisionerState` value and have the tracking be done within the method itself.

Reply:
Created an issue for the 3rd issue: [#1655](https://github.com/linkedin/cruise-control/issues/1655)